src/main/java/seedu/meal360/Parser.java

```diff
@@ -0,0 +1,641 @@
+package seedu.meal360;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+import seedu.meal360.exceptions.IngredientNotFoundException;
+import seedu.meal360.exceptions.InvalidNegativeValueException;
+import seedu.meal360.exceptions.InvalidRecipeNameException;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+public class Parser {
+
```

Comment:
Seems like the parser class is doing too much (ie. handling all forms of commands within itself). Perhaps consider adding sub classes for handling each important group of command so that it reduces the functionality of parser and each sub class can be independently handled easily